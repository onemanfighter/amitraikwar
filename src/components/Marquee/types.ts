import { BoxProps } from '@chakra-ui/react';
import { ComponentPropsWithoutRef } from 'react';

/**
 * Props for the Marquee component
 * @extends ComponentPropsWithoutRef<'div'>
 * @property {boolean} [reverse=false] - Whether to reverse the animation direction
 * @property {boolean} [pauseOnHover=false] - Whether to pause the animation on hover
 * @property {React.ReactNode} children - Content to be displayed in the marquee
 * @property {boolean} [vertical=false] - Whether to animate vertically instead of horizontally
 * @property {number} [repeat=4] - Number of times to repeat the content
 */
export interface MarqueeProps extends ComponentPropsWithoutRef<'div'> {
  reverse?: boolean;
  pauseOnHover?: boolean;
  children: React.ReactNode;
  vertical?: boolean;
  repeat?: number;
  duration?: number;
  gap?: BoxProps['gap'];
}
