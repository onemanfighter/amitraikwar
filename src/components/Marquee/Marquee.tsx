import { Box } from '@chakra-ui/react';
import { MarqueeProps } from './types';
import { useState } from 'react';

const Marquee = ({
  reverse = false,
  pauseOnHover = true,
  vertical = false,
  repeat = 4,
  duration = 10,
  children,
  gap = 4,
}: MarqueeProps) => {
  const [isHovering, setHovering] = useState(false);
  const hoverProps = pauseOnHover
    ? {
        onMouseEnter: () => setHovering(true),
        onMouseLeave: () => setHovering(false),
      }
    : {};
  return (
    <Box
      style={{
        display: 'flex',
        overflow: 'hidden',
        padding: '0.5rem', // Assuming p-2 translates to 0.5rem padding
        flexDirection: vertical ? 'column' : 'row',
      }}
      gap={gap}
      {...hoverProps}
    >
      <style>
        {`
          @keyframes marquee {
            0% { transform: translateX(0%); }
            100% { transform: translateX(-100%); }
          }
          @keyframes marquee-vertical {
            0% { transform: translateY(0%); }
            100% { transform: translateY(-100%); }
          }
        `}
      </style>
      {Array(repeat)
        .fill(0)
        .map((_, i) => (
          <Box
            key={i}
            gap={gap}
            display={'flex'}
            flexShrink={0}
            justifyContent={'space-around'}
            animation={
              (vertical
                ? `marquee-vertical ${duration}s linear infinite`
                : `marquee ${duration}s linear infinite`) +
              (isHovering ? ' paused ' : ' running ') +
              (reverse ? 'reverse' : 'normal')
            }
          >
            {children}
          </Box>
        ))}
    </Box>
  );
};

export default Marquee;
