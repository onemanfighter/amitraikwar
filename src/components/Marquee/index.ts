export { default as Marquee } from './Marquee';
export * from './types';
