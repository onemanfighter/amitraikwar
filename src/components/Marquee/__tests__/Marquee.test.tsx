import { fireEvent, render } from '@testing-library/react';
import Marquee from '../Marquee';

describe('Marquee', () => {
  it('should render the Marquee component', () => {
    const { container } = render(<Marquee>Test container</Marquee>);

    expect(container).toMatchSnapshot();
  });

  it('should render the Marquee component', () => {
    const { container } = render(
      <Marquee
        repeat={40}
        duration={10}
        gap={4}
        pauseOnHover={true}
        reverse={true}
      >
        Test container
      </Marquee>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the Marquee component with vertical direction', () => {
    const { container } = render(
      <Marquee
        repeat={40}
        duration={10}
        gap={4}
        pauseOnHover={true}
        reverse={true}
        vertical={true}
      >
        Test container
      </Marquee>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the Marquee component with horizontal direction', () => {
    const { container } = render(
      <Marquee
        repeat={40}
        duration={10}
        gap={4}
        pauseOnHover={true}
        reverse={true}
        vertical={false}
      >
        Test container
      </Marquee>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the Marquee component without hover', () => {
    const { container } = render(
      <Marquee
        repeat={40}
        duration={10}
        gap={4}
        pauseOnHover={false}
        reverse={true}
        vertical={false}
      >
        Test container
      </Marquee>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should render the Marquee component with reverse false', () => {
    const { container } = render(
      <Marquee
        repeat={40}
        duration={10}
        gap={4}
        pauseOnHover={false}
        reverse={false}
        vertical={false}
      >
        Test container
      </Marquee>,
    );

    expect(container).toMatchSnapshot();
  });
});
