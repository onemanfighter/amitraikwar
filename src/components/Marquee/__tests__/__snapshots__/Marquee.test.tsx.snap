// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marquee should render the Marquee component 1`] = `
<div>
  <div
    class="css-lhfv11"
    style="display: flex; overflow: hidden; padding: 0.5rem; flex-direction: row;"
  >
    <style>
      
          @keyframes marquee {
            0% { transform: translateX(0%); }
            100% { transform: translateX(-100%); }
          }
          @keyframes marquee-vertical {
            0% { transform: translateY(0%); }
            100% { transform: translateY(-100%); }
          }
        
    </style>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
  </div>
</div>
`;

exports[`Marquee should render the Marquee component 2`] = `
<div>
  <div
    class="css-lhfv11"
    style="display: flex; overflow: hidden; padding: 0.5rem; flex-direction: row;"
  >
    <style>
      
          @keyframes marquee {
            0% { transform: translateX(0%); }
            100% { transform: translateX(-100%); }
          }
          @keyframes marquee-vertical {
            0% { transform: translateY(0%); }
            100% { transform: translateY(-100%); }
          }
        
    </style>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
  </div>
</div>
`;

exports[`Marquee should render the Marquee component with horizontal direction 1`] = `
<div>
  <div
    class="css-lhfv11"
    style="display: flex; overflow: hidden; padding: 0.5rem; flex-direction: row;"
  >
    <style>
      
          @keyframes marquee {
            0% { transform: translateX(0%); }
            100% { transform: translateX(-100%); }
          }
          @keyframes marquee-vertical {
            0% { transform: translateY(0%); }
            100% { transform: translateY(-100%); }
          }
        
    </style>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
  </div>
</div>
`;

exports[`Marquee should render the Marquee component with reverse false 1`] = `
<div>
  <div
    class="css-lhfv11"
    style="display: flex; overflow: hidden; padding: 0.5rem; flex-direction: row;"
  >
    <style>
      
          @keyframes marquee {
            0% { transform: translateX(0%); }
            100% { transform: translateX(-100%); }
          }
          @keyframes marquee-vertical {
            0% { transform: translateY(0%); }
            100% { transform: translateY(-100%); }
          }
        
    </style>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
    <div
      class="css-15jpxfq"
    >
      Test container
    </div>
  </div>
</div>
`;

exports[`Marquee should render the Marquee component with vertical direction 1`] = `
<div>
  <div
    class="css-lhfv11"
    style="display: flex; overflow: hidden; padding: 0.5rem; flex-direction: column;"
  >
    <style>
      
          @keyframes marquee {
            0% { transform: translateX(0%); }
            100% { transform: translateX(-100%); }
          }
          @keyframes marquee-vertical {
            0% { transform: translateY(0%); }
            100% { transform: translateY(-100%); }
          }
        
    </style>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
    <div
      class="css-3si7d9"
    >
      Test container
    </div>
  </div>
</div>
`;

exports[`Marquee should render the Marquee component without hover 1`] = `
<div>
  <div
    class="css-lhfv11"
    style="display: flex; overflow: hidden; padding: 0.5rem; flex-direction: row;"
  >
    <style>
      
          @keyframes marquee {
            0% { transform: translateX(0%); }
            100% { transform: translateX(-100%); }
          }
          @keyframes marquee-vertical {
            0% { transform: translateY(0%); }
            100% { transform: translateY(-100%); }
          }
        
    </style>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
    <div
      class="css-19uklyi"
    >
      Test container
    </div>
  </div>
</div>
`;
